apps/server/src/controllers/score.ts

```diff
@@ -0,0 +1,191 @@
+import { IMatch } from '@repo/shared-types';
+import { Request, Response } from 'express';
+
+import User from '../models/users';
+
+export const addScoreAndMatch = async (req: Request, res: Response) => {
+  try {
+    const { username } = req.params;
+    const { score, match }: { score: number; match: IMatch } = req.body;
+
+    if (typeof score !== 'number' || score <= 0) {
+      return res.status(400).json({ message: 'Invalid Score' });
+    }
+    if (!match.opponent || !match.outcome) {
+      return res.status(400).json({ message: 'Invalid opponent or outcome' });
+    }
+
+    const updatedUser = await User.findOneAndUpdate(
+      { username: username },
+      {
+        $inc: {
+          total_score: score,
+          games_won: match.outcome === 'win' ? 1 : 0,
+          games_lost: match.outcome === 'lose' ? 1 : 0,
+          games_draw: match.outcome === 'draw' ? 1 : 0,
+        },
+        $push: { matchHistory: match },
+      },
+      { new: true }
+    );
+
+    if (!updatedUser) {
+      return res.status(404).json({ message: 'User not found.' });
```

Comment:
I think 404 is reserve for when then endpoints or the pages requests is not available. I think we can use 400 as well (open to discussion as well)